streams/src/test/java/org/apache/kafka/streams/processor/internals/tasks/DefaultTaskExecutorTest.java

```diff
@@ -85,6 +85,16 @@ public void shouldShutdownTaskExecutor() {
         assertNull(taskExecutor.currentTask(), "Have task assigned after shutdown");
     }
 
+    @Test
+    public void shouldAwaitProcessableTasksIfNoneAssignable() throws InterruptedException {
+        assertNull(taskExecutor.currentTask(), "Have task assigned before startup");
+        when(task.isProcessable(anyLong())).thenReturn(false);
+
+        taskExecutor.start();
+
+        verify(taskManager, timeout(VERIFICATION_TIMEOUT).atLeastOnce()).awaitProcessableTasks();
+    }
+
```

Comment:
This test is a bit hard to follow. I think the test would be easier to understand if you specify the simpler stub `when(taskManager.assignNextTask(taskExecutor)).thenReturn(null);` here in the test instead of using the stub specified in `setUp()`.

Reply:
Done